nservicebus/hosting/cloud-services-host/configuration.md

```diff
@@ -1,114 +1,78 @@
-xmlns="http://schemas.microsoft.com/ServiceHosting/2008/10/ServiceConfiguration"
-schemaVersion="2013-03.2.0">
-  <Role name="{role name}">
-    <Instances count="1" />
-    <ConfigurationSettings>
-      <Setting name="AzureQueueConfig.ConnectionString"
-      value="DefaultEndpointsProtocol=https;AccountName={the account name here};
-      AccountKey={the account key here}" />
-      <Setting name="AzureQueueConfig.QueueName" value="{the queue name here}" />
-...
-    </ConfigurationSettings>
-  </Role>
-</ServiceConfiguration>
-```
-
-The "AzureQueueConfig.ConnectionString" for Azure Queues follows this format:
-
-    DefaultEndpointsProtocol=https;AccountName=myAccount;AccountKey=myKey;QueueEndpoint=customEndpoint;
-
-Alternatively, use the Azure development environment emulator by using this connection string:
-
-    UseDevelopmentStorage=True;
-
-In the NServiceBus solution, specify the Endpoint Configuration to use AzureStorageQueue transport:
-
-```
-public class EndpointConfig : IConfigureThisEndpoint, AsA_Worker
-{
-    public void Customize(BusConfiguration builder)
-    {
-        builder.UseTransport<AzureStorageQueueTransport>();
-    }
-}
-```
-
-### Azure Service Bus
-
-In the Azure Service Configuration file
-(ServiceConfiguration.cscfg), add the following sections:
-
-```
-<?xml version="1.0" encoding="utf-8"?>
-<ServiceConfiguration serviceName="AzureService"
-xmlns="http://schemas.microsoft.com/ServiceHosting/2008/10/ServiceConfiguration"
-schemaVersion="2013-03.2.0">
-  <Role name="{role name}">
-    <Instances count="1" />
-    <ConfigurationSettings>
-      <Setting name="AzureServiceBusQueueConfig.ConnectionString"
-      value="Endpoint=sb://{namespace here}.servicebus.windows.net/;
-      SharedAccessKeyName=RootManageSharedAccessKey;
-      SharedAccessKey={shared access key here}" />
-...
-    </ConfigurationSettings>
-  </Role>
-</ServiceConfiguration>
-```
-
-
-The "AzureServiceBusQueueConfig.ConnectionString" for Azure Service Bus namespace connection string can be retrieved from the Azure portal using an authorized account.
-
-In the NServiceBus solution, specify the endpoint configuration to use AzureServiceBus transport:
-
-```
-public class EndpointConfig : IConfigureThisEndpoint, AsA_Worker
-{
-    public void Customize(BusConfiguration builder)
-    {
-        builder.UseTransport<AzureServiceBusTransport>();
-    }
-}
-```
\ No newline at end of file
+## Configuration override convention
+
+Because Azure cloud services has it's own configuration model, but NServiceBus is typically used with it's configuration in app.config, a convention based configuration source is provided. Where most of the configuration is in app.config, but any setting can be added 'by convention' to the service configuration file to override the original value in app.config. This makes it easy to develop locally (without the service runtime), but still make use of this feature in production.
```

Comment:
NServiceBus is typically configured using an `app.config` file, however Azure Cloud Services have their own configuration model. That makes settings management between various environments complicated. In order to simplify the process, NServiceBus also supports a convention-based configuration. It allows for adding any NServiceBus setting to the service configuration file. The value specified in the service configuration file will override the value specified in the `app.config` file.